### ISSUE TITLE
[bug] useContractReads returning [null] or undefined data unexpectedly

### ISSUE THREAD

**Issue by matthewlilley** [tmm/wagmi#645]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

latest

### Current Behavior

There was a previous issue which was closed but had reported the same issue. Maybe there is some human error here but I'm struggling to find it. I've added a repro to illustrate this one with the same data.

### Expected Behavior

Work

### Steps To Reproduce

repro attached

### Link to Minimal Reproducible Example (CodeSandbox, StackBlitz, etc.)

https://codesandbox.io/s/pensive-danny-fzwj5g?file=/src/App.js

### Anything else?

We are migrating a large amount of our codebase from redux multicall to wagmi multicall, but have run into a number of issues so far so we are halting and will help illustrate them until there is more stability.

An example of this hook migration from redux multicall to wagmi multicall.

![Screenshot_272](https://user-images.githubusercontent.com/6063350/176894243-20fdc0df-6b90-4bdd-b495-a3e4665e8340.png)

**Comment by tmm** [comment]
After some inspecting, I'm seeing the following error (when setting [`allowFailure`](https://wagmi.sh/docs/hooks/useContractReads#allowfailure-optional=) to `true`)

```
call revert exception [ See: https://links.ethers.org/v5-errors-CALL_EXCEPTION ] (method="getReserves()", data="0x", errorArgs=null, errorName=null, errorSignature=null, reason=null, code=CALL_EXCEPTION, version=abi/5.6.1)
```

I tried running the hook with a smaller number of contract addresses and discovered that (at least) the following are EOAs (I would double check the other addresses):

- [`0x9dB53AD5611ea193D58Ea193f89e984B27E98Dd8`](https://snowtrace.io/address/0x9dB53AD5611ea193D58Ea193f89e984B27E98Dd8)
- [`0xB0159c366F3e3B9a76A827206419062051A2e7Eb`](https://snowtrace.io/address/0xB0159c366F3e3B9a76A827206419062051A2e7Eb)
- [`0x977B8A43D70Fbe6A409F499F1B92Ea24f3c6D01e`](https://snowtrace.io/address/0x977B8A43D70Fbe6A409F499F1B92Ea24f3c6D01e)
- [`0x53480aAc4dDe9C5380D20B693f02932535F6409d`](https://snowtrace.io/address/0x53480aAc4dDe9C5380D20B693f02932535F6409d)

Once the EOAs are removed, everything worked great!

**Comment by matthewlilley** [comment]
@tmm Ah, these aren't nessasarily EOAs, but they're precomputed (using create2 methodology) addresses of potential pairs. There's no way to know if these pairs exist or not before hand, so it's important that multicalls don't fail as a whole and can independently fail. It's worth looking at the use case here:

https://github.com/sushiswap/sushiswap/blob/master/apps/swap/lib/hooks/usePairs.ts#L81

**Comment by matthewlilley** [comment]
Might be useful to look at for this one: https://github.com/Uniswap/redux-multicall/blob/ab054d50e777b472e1f2380d33d22a3df6ff2d1a/src/utils/callState.ts#L21

**Comment by jxom** [comment]
Pushed up a fix at #654

**Comment by matthewlilley** [comment]
@jxom Yeah, this appears to act the way I expect now in the example. Good job!

https://codesandbox.io/s/pensive-danny-fzwj5g?file=/src/App.js
